### PR TITLE
BUG: loc with empty multiindex raises exception

### LINKED ISSUE #36936: BUG: loc with empty multiindex raises exception
- [x] I have checked that this issue has not already been reported.

- [x] I have confirmed this bug exists on the latest version of pandas.

- [ ] (optional) I have confirmed this bug exists on the master branch of pandas.

---
Hi, last line of snippet throws an exception ``ValueError: operands could not be broadcast together with shapes (0,) (2,) (0,)``

```python
import pandas as pd

arrays = [['val1', 'val1', 'val2'], ['val1', 'val1', 'val2']]
index = pd.MultiIndex.from_arrays(arrays, names=('idx1', 'idx2'))

df = pd.DataFrame([1, 2, 3], index=index, columns=['value'])

df.loc[df[df['value'] == 0].index, :] = 4

```

#### Problem description

``.loc`` raises error in case dataframe has duplicates in index. ``df[df['value'] == 0].index`` produces ``MultiIndex([], names=['idx1', 'idx2'])`` and it seems to be absolutely valid to put it into ``.loc`` method.

#### Expected Output

I expect ``.loc`` operation to make no changes to original ``df`` cause multiindex in ``.loc`` is empty.

#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit           : db08276bc116c438d3fdee492026f8223584c477
python           : 3.7.6.final.0
python-bits      : 64
OS               : Windows
OS-release       : 10
Version          : 10.0.17763
machine          : AMD64
processor        : Intel64 Family 6 Model 142 Stepping 9, GenuineIntel
byteorder        : little
LC_ALL           : None
LANG             : None
LOCALE           : None.None

pandas           : 1.1.3
numpy            : 1.19.1
pytz             : 2020.1
dateutil         : 2.8.1
pip              : 20.2.2
setuptools       : 46.0.0
Cython           : None
pytest           : 6.0.1
hypothesis       : None
sphinx           : None
blosc            : None
feather          : None
xlsxwriter       : 1.3.0
lxml.etree       : None
html5lib         : None
pymysql          : None
psycopg2         : 2.8.5 (dt dec pq3 ext lo64)
jinja2           : 2.11.1
IPython          : 7.14.0
pandas_datareader: None
bs4              : 4.9.1
bottleneck       : None
fsspec           : None
fastparquet      : None
gcsfs            : None
matplotlib       : 3.2.1
numexpr          : None
odfpy            : None
openpyxl         : 3.0.4
pandas_gbq       : None
pyarrow          : 1.0.1
pytables         : None
pyxlsb           : 1.0.6
s3fs             : None
scipy            : 1.4.1
sqlalchemy       : 1.3.17
tables           : None
tabulate         : None
xarray           : None
xlrd             : 1.2.0
xlwt             : 1.3.0
numba            : None

</details>

### PR DESCRIPTION
**Pull request by kasim95**
- [x] closes #36936 
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

### CODE REVIEW

**Comment by kasim95** [comment]
Added test case for the changes made in GH#36936

File: pandas/tests/indexing/multiindex/test_loc.py [hunk 1]
```diff
@@ -695,3 +695,22 @@ def test_loc_getitem_index_differently_ordered_slice_none():
...
+    empty_multiindex = df.loc[df.loc[:, "value"] == 0, :].index
+    result = df.loc[empty_multiindex, :]
+    expected = df.loc[[False] * len(df.index), :]
+    tm.assert_equal(result, expected)
+
+    # replacing value with loc on empty multiindex
+    df.loc[df.loc[df.loc[:, "value"] == 0].index, "value"] = 5
+    result = df
+    expected = DataFrame([1, 2, 3, 4], index=index, columns=["value"])
+    tm.assert_equal(result, expected)
```
Comment (MarcoGorelli) [inline pandas/tests/indexing/multiindex/test_loc.py:716]: does `tm.assert_frame_equal` not work here?

File: pandas/tests/indexing/multiindex/test_loc.py [hunk 2]
```diff
@@ -695,3 +695,22 @@ def test_loc_getitem_index_differently_ordered_slice_none():
...
+def test_loc_empty_multiindex():
+    # GH#36936
+    arrays = [["a", "a", "b", "a"], ["a", "a", "b", "b"]]
+    index = MultiIndex.from_arrays(arrays, names=("idx1", "idx2"))
+    df = DataFrame([1, 2, 3, 4], index=index, columns=["value"])
+
+    # loc on empty multiindex == loc with False mask
+    empty_multiindex = df.loc[df.loc[:, "value"] == 0, :].index
+    result = df.loc[empty_multiindex, :]
+    expected = df.loc[[False] * len(df.index), :]
```
Comment (jreback) [inline pandas/tests/indexing/multiindex/test_loc.py:709]: can you construct the expected frame explicityly

File: pandas/tests/indexing/multiindex/test_loc.py [hunk 3]
```diff
@@ -695,3 +695,22 @@ def test_loc_getitem_index_differently_ordered_slice_none():
         columns=["a", "b"],
     )
     tm.assert_frame_equal(result, expected)
+
+
+def test_loc_empty_multiindex():
```
Comment (jreback) [inline pandas/tests/indexing/multiindex/test_loc.py:700]: can you co-locate this with similar tests

```
jreback@lhs2:~/pandas-dev$ grep empty pandas/tests/indexing/multiindex/*
pandas/tests/indexing/multiindex/test_getitem.py:def test_frame_getitem_multicolumn_empty_level():
pandas/tests/indexing/multiindex/test_getitem.py:def test_frame_mi_empty_slice():
pandas/tests/indexing/multiindex/test_loc.py:        empty = Series(data=[], dtype=np.float64)
pandas/tests/indexing/multiindex/test_loc.py:        result = x.loc[empty]
pandas/tests/indexing/multiindex/test_loc.py:        # empty array:
pandas/tests/indexing/multiindex/test_loc.py:        empty = np.array([])
pandas/tests/indexing/multiindex/test_loc.py:        result = x.loc[empty]
pandas/tests/indexing/multiindex/test_loc.py:        ([], []),  # empty ok
pandas/tests/indexing/multiindex/test_loc.py:def test_loc_getitem_duplicates_multiindex_empty_indexer(columns_indexer):
pandas/tests/indexing/multiindex/test_loc.py:    # empty indexer
```

IOW there are 2 tests in the test_getitem.py that are testing basically the same, so locate there.

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tests/indexing/multiindex/test_loc.py (hunk 1 above)
Comment (kasim95) [reply pandas/tests/indexing/multiindex/test_loc.py]: The assertion method is changed to `tm.assert_frame_equal` in the most recent commit.

File: pandas/tests/indexing/multiindex/test_loc.py (hunk 3 above)
Comment (kasim95) [reply pandas/tests/indexing/multiindex/test_loc.py]: Moved the test to *pandas/tests/indexing/multiindex/test_getitem.py*

File: pandas/tests/indexing/multiindex/test_loc.py (hunk 2 above)
Comment (kasim95) [reply pandas/tests/indexing/multiindex/test_loc.py]: Constructing an empty DataFrame named `expected` with Multiindex assigns the `expected.index.inferred_type` attribute to `object` by default. 
The value of `result.index.inferred_type` attribute is `string` as inferred from the `index` variable created earlier.
This fails the `tm.assert_frame_equal` test

I used the following code to create an empty dataframe with multiindex:
```python
def test_loc_empty_multiindex():
    # GH#36936
    arrays = [["a", "a", "b", "a"], ["a", "a", "b", "b"]]
    index = MultiIndex.from_arrays(arrays, names=("idx1", "idx2"))
    df = DataFrame([1, 2, 3, 4], index=index, columns=["value"])

    # loc on empty multiindex == loc with False mask
    empty_multiindex = df.loc[df.loc[:, "value"] == 0, :].index
    result = df.loc[empty_multiindex, :]
    index2 = MultiIndex(levels=[[], []], codes=[[], []], names=["idx1", "idx2"])
    expected = DataFrame([], index=index2, columns=["value"], dtype="string")
    expected = expected.astype({"value": "int"})
    tm.assert_frame_equal(result, expected) # this test fails

    # replacing value with loc on empty multiindex
    df.loc[df.loc[df.loc[:, "value"] == 0].index, "value"] = 5
    result = df
    expected = DataFrame([1, 2, 3, 4], index=index, columns=["value"])
    tm.assert_frame_equal(result, expected)
```

If the `index` variable is used directly to create the empty dataframe, it introduces NaN values

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @kasim95

**Pull request merged**